### PR TITLE
Add optional scaling preprocessing to prediction  

### PR DESCRIPTION
**Pull request by maweigert**
This PR adds a scalar scale parameter to `model.predict_instances` that scales the input before prediction and rescales the output accordingly to accommodate varying object sizes in the input. 

```python
from csbdeep.utils import normalize
from stardist.data import test_image_nuclei_2d
from stardist.models import StarDist2D

model = StarDist2D.from_pretrained('2D_versatile_fluo')
x = test_image_nuclei_2d()
x = normalize(x)
labels, res = model.predict_instances(x, scale=0.6, verbose=True)

```

### CODE REVIEW

File: stardist/models/model2d.py [hunk 1]
```diff
@@ -512,6 +512,10 @@ def _instances_from_prediction(self, img_shape, prob, dist, points=None, prob_cl
                 inds = tuple(p//g for p,g in zip(points.T, self.config.grid))
                 prob_class = prob_class[inds]
 
+        if scale is not None:
+            points = points/scale
+            disti = disti/scale
+
         if return_labels:
             labels = polygons_to_label(disti, points, prob=probi, shape=img_shape)
```
Comment (uschmidt83) [inline stardist/models/model2d.py:520]: Can't do multiple scale values (one per axis) because `polygons_to_label` uses distances, and not coordinates.

Comment (uschmidt83) [reply stardist/models/model2d.py:520]: We could easily update `polygons_to_label` to also take `scale` as an input, but it's not so easy in 3D, where the C++ code needs to be updated.

Comment (uschmidt83) [reply stardist/models/model2d.py:520]: I think that makes sense, because a one might not want to scale the `Z` axis in the same way as `X` and `Y`.

Comment (uschmidt83) [reply stardist/models/model2d.py:520]: I've already implemented per axis scale values in `predict_instances` and could pass a dict (axis -> scale) on to `__instances_from_prediction`.

Comment (uschmidt83) [reply stardist/models/model2d.py:520]: I could modify the 3D C++ code, but I might screw it up somewhere ;)

Comment (maweigert) [reply stardist/models/model2d.py:520]: sounds good!

Comment (maweigert) [reply stardist/models/model2d.py:520]: > I could modify the 3D C++ code, bu

Yes, please don't :)

**Pull request merged**